python-threatexchange/threatexchange/cli/match_cmd.py

```diff
@@ -167,19 +167,22 @@ def execute(self, settings: CLISettings) -> None:
         if self.as_hashes:
             types = (BytesHasher, TextHasher, FileHasher)
         signal_types = [s for s in signal_types if issubclass(s, types)]
```

Comment:
(In reference to below comments) This list of signal types have already been narrowed down the valid set given the input.

It might be that we should consider moving `--hashes` to a toplevel argument (tx match photo|video|hash)` and allowing cross matching might make more sense, but now we've exceeded the scope of the pr.